backend/controller/observability/observability.go

```diff
@@ -0,0 +1,207 @@
+package observability
+
+import (
+	"context"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"time"
+
+	"github.com/alecthomas/types/optional"
+
+	"github.com/TBD54566975/ftl/backend/controller/dal"
+	ftlv1 "github.com/TBD54566975/ftl/backend/protos/xyz/block/ftl/v1"
+	"github.com/TBD54566975/ftl/backend/schema"
+	"github.com/TBD54566975/ftl/internal/log"
+	"github.com/TBD54566975/ftl/internal/model"
+)
+
+const name = "ftl.xyz/ftl/runner"
+
+type metricAttributeBuilders struct {
+	moduleName      func(name string) attribute.KeyValue
+	featureName     func(name string) attribute.KeyValue
+	destinationVerb func(name string) attribute.KeyValue
+}
+
+type callMetrics struct {
+	requests metric.Int64Counter
+	failures metric.Int64Counter
+	active   metric.Int64UpDownCounter
+	latency  metric.Int64Histogram
+}
+
+type fsmMetrics struct {
+	active      metric.Int64UpDownCounter
+	transitions metric.Int64Counter
+	failures    metric.Int64Counter
+}
+
+type observableMetrics struct {
+	meter      metric.Meter
+	attributes metricAttributeBuilders
+	calls      *callMetrics
+	fsm        *fsmMetrics
+}
+
+var (
+	metrics = observableMetrics{
+		meter: otel.Meter(name),
+		// TODO: move to a initialization method
+		attributes: metricAttributeBuilders{
+			moduleName: func(name string) attribute.KeyValue {
+				return attribute.String("ftl.module.name", name)
+			},
+			featureName: func(name string) attribute.KeyValue {
+				return attribute.String("ftl.feature.name", name)
+			},
+			destinationVerb: func(name string) attribute.KeyValue {
+				return attribute.String("ftl.verb.dest", name)
+			},
+		},
+		calls: &callMetrics{},
+		fsm:   &fsmMetrics{},
+	}
+)
+
+type CallBegin struct {
+	DestVerb *schema.Ref
+	Callers  []*schema.Ref
+}
+
+type CallEnd struct {
+	DeploymentKey model.DeploymentKey
+	RequestKey    model.RequestKey
+	StartTime     time.Time
+	DestVerb      *schema.Ref
+	Callers       []*schema.Ref
+	Request       *ftlv1.CallRequest
+	Response      optional.Option[*ftlv1.CallResponse]
+	CallError     optional.Option[error]
+}
+
+func init() {
+	metrics.calls.requests, _ = metrics.meter.Int64Counter("ftl.call.requests",
+		metric.WithDescription("number of verb calls"),
+		metric.WithUnit("{count}"))
```

Comment:
The unit names all have to come from here so I'm pretty sure we have to leave the counts empty: https://ucum.org/ucum#section-Alphabetic-Index-By-Name